### PR TITLE
Prefer songs to albums

### LINKED ISSUE #270: Prefer songs to albums when both exist with the same name
From the logs:
```
U: play yellow brick road by elton john
UT: $dialogue @org.thingpedia.dialogue.transaction.execute;
UT: @com.spotify.playable() filter contains(artists, null^^com.spotify:artist("elton john")) && id =~ "yellow brick road" => @com.spotify.play(playable=id);
C: $dialogue @org.thingpedia.dialogue.transaction.execute;
C: (@com.spotify(id="com.spotify-gcampax"^^tt:device_id("Spotify Account of gcampax")).playable() filter contains(artists, "spotify:artist:3PhoLpVuITZKcymswpck5b"^^com.spotify:artist("Elton John")) && id =~ "yellow brick road")[1] => @com.spotify(id="com.spotify-gcampax"^^tt:device_id("Spotify Account of gcampax")).play(playable=id)
C: #[results=[
C:   { device="b305db3b61d1c95661290444e9ddef901ccfff60"^^com.spotify:device("giovanni-xps15"), playable="spotify:album:5WupqgR68HfuHt3BMJtgun"^^com.spotify:playable("Goodbye Yellow Brick Road (Remastered)") }
C: ]];
A: I played Goodbye Yellow Brick Road (Remastered) on Spotify.
AT: $dialogue @org.thingpedia.dialogue.transaction.sys_action_success;
```

It played the album (which starts with a different song) and not the song. This is confusing, and makes it seem like the agent is misbehaving.

### PR DESCRIPTION
**Pull request by ryachen01**
Fixes #270

### CODE REVIEW

**Review by gcampax — COMMENTED** [review]
Should we add a scenario test for the specific case that caused this bug? So we don't break it in the future.

**Comment by ryachen01** [comment]
Partially fixes #280 
Are there any other known cases of bad error messages?

**Comment by gcampax** [comment]
> Partially fixes #280
> Are there any other known cases of bad error messages?

I don't know to be honest. I think we should log the detailed output of all errors so we can catch more cases in the future. I see the Spotify API is finnicky and doesn't return nice errors.

File: main/com.spotify/manifest.tt [hunk 1]
```diff
@@ -254,7 +254,8 @@ class @com.spotify
     no_active_device=["you must open the spotify app first", "no spotify player is active"],
     disallowed_action=["failed to play: you must first click play from inside the spotify app , and then i will play songs for you"],
     player_error=["failed to play: you must first click play from inside the spotify app , and then i will play songs for you"],
-    dailymix_error=["spotify doesn't support daily mix playlists at this time"]
+    dailymix_error=["spotify doesn't support daily mix playlists at this time"],
+    non_premium_account=["you need a spotify premium account to do this"]
```
Comment (gcampax) [inline main/com.spotify/manifest.tt:258]: how about "you need a spotify premium account to play music"

**Review by gcampax — APPROVED** [review]
Yeah let's merge this for now (fix the tests), and let's see what other errors we catch.

**Comment by ryachen01** [comment]
The problem with all the errors mapping to disallowed action should be fixed now. What was happening was that the hooks were being set and then a bunch of errors were being thrown, but we only showed the last error which was always a disallowed action from _flushPlay

**Comment by ryachen01** [comment]
Also, I think the manifest needs to be updated in order for the scenario tests to pass

**Comment by gcampax** [comment]
Mh this will cause issues for anyone who already has configured Spotify and is testing. Can we treat missing product field ("undefined") the same as premium?

**Pull request merged**